pyuvdata/uvdata.py

```diff
@@ -1185,6 +1185,144 @@ def __iadd__(self, other):
         self.__add__(other, inplace=True)
         return self
 
+    def fast_concat(self, other, axis, run_check=True, check_extra=True,
+                    run_check_acceptability=True, inplace=False):
+        """
+        Concatenate two UVData objects along specified axis with almost no checking of metadata.
+
+        Warning! This method assumes all the metadata along other axes is sorted
+        the same way. The __add__ method is much safer, it checks all the metadata,
+        but it is slower. Some quick checks are run, but this method doesn't
+        make any guarantees that the resulting object is correct.
+
+        Args:
+            other: Another UVData object which will be added to self.
+            axis: Axis to concatenate files along. This enables fast concatenation
+                along the specified axis without the normal checking that all other
+                metadata agrees. Allowed values are: 'blt', 'freq', 'polarization'.
+            run_check: Option to check for the existence and proper shapes of
+                parameters after combining objects. Default is True.
+            check_extra: Option to check optional parameters as well as
+                required ones. Default is True.
+            run_check_acceptability: Option to check acceptable range of the values of
+                parameters after combining objects. Default is True.
+            inplace: Overwrite self as we go, otherwise create a third object
+                as the sum of the two (default).
+        """
+        if inplace:
+            this = self
+        else:
+            this = copy.deepcopy(self)
+        # Check that both objects are UVData and valid
+        this.check(check_extra=check_extra, run_check_acceptability=run_check_acceptability)
+        if not issubclass(other.__class__, this.__class__):
+            if not issubclass(this.__class__, other.__class__):
+                raise ValueError('Only UVData (or subclass) objects can be '
+                                 'added to a UVData (or subclass) object')
+        other.check(check_extra=check_extra, run_check_acceptability=run_check_acceptability)
+
+        allowed_axes = ['blt', 'freq', 'polarization']
+        if axis is not None:
```

Comment:
Looks like `axis` is required, so I'm not sure this `if` is necessary.

Reply:
yep, thanks!